ops/charm.py

```diff
@@ -823,77 +826,105 @@ def defer(self):
 class CharmEvents(ObjectEvents):
     """Events generated by Juju pertaining to application lifecycle.
 
-    This class is used to create an event descriptor (``self.on``) attribute for
+    This class is used to create an event descriptor (``self.on``) for
```

Comment:
I'd rather give a more 'actionable' usage hint.
How about:

"subclass this type to achieve X, Y and Z. For example:..."

Reply:
I've actually removed this paragraph as I think it's misleading here. `CharmBase` already has an `on` so you don't need to mess with it unless you have custom events (but that should probably be documented on `ObjectEvents`).